src/wallet/rpcdump.cpp

```diff
@@ -1000,7 +1001,7 @@ static UniValue ProcessImport(CWallet * const pwallet, const UniValue& data, con
             const auto& str = keys[i].get_str();
             CKey key = DecodeSecret(str);
             if (!key.IsValid()) {
-                throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Invalid private key encoding");
+                throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Invalid private key encoding at index " + i);
```

Comment:
This looks like very dangerous to me :-)

Reply:
@practicalswift why?

Reply:
@promag You're talking a pointer to a char array and adding an integer to it. If i is larger than the number of characters in the string, this results in reading in a buffer overflow.

Reply:
Oops, thinking too pythonically. Fixed thanks :)

Reply:
God 🤦‍♂️